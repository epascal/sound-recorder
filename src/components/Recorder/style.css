main button.record-play {
    align-self: center;
    background-color: rgb(14, 60, 14);
    color: white;
    font-weight: 700;
    height: 50px;
    width: 150px;
    margin-top: 10px;
}

main button.record-play.recording-audio {
    background-color: tomato;
    border-color: rgb(206, 74, 51);
    color: white;
}
  