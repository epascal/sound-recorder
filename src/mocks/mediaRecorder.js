Object.defineProperty(window, 'MediaRecorder', {
    writable: true,
    value: jest.fn().mockImplementation(() => ({
        start: jest.fn(),
        ondataavailable: jest.fn(),
        onerror: jest.fn(),
        state: "",
        stop: jest.fn(),
        pause: jest.fn(),
        resume: jest.fn(),
    })),
})

Object.defineProperty(MediaRecorder, 'isTypeSupported', {
    writable: true,
    value: () => true
})


// export const mockMediaRecorder = jest.fn();
// const mock = jest.fn().mockImplementation(() => {
//     return {
//         playSoundFile: mockMediaRecorder
//     };
// });


// export default mock