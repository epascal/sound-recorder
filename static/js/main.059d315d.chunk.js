(this["webpackJsonpsound-recorder"]=this["webpackJsonpsound-recorder"]||[]).push([[0],[,,,,,,,,,,,function(e,t,n){},,function(e,t,n){},,function(e,t,n){},function(e,t,n){},function(e,t,n){"use strict";n.r(t);var r=n(1),c=n.n(r),o=n(5),a=n.n(o),i=(n(11),n(3)),s=n.n(i),u=n(6),d=n(4),l=(n(13),n(0)),j=function(e){var t=e.stream;return Object(l.jsx)(l.Fragment,{children:Object(l.jsxs)("article",{children:[Object(l.jsx)("audio",{controls:"controls",src:t}),Object(l.jsx)("p",{children:"Unnamed recording"}),Object(l.jsx)("button",{onClick:function(){},children:"Delete"})]},t.toString())})},b=(n(15),function(){var e=Object(r.useState)("Record"),t=Object(d.a)(e,2),n=t[0],c=t[1],o=Object(r.useState)([]),a=Object(d.a)(o,2),i=a[0],b=a[1],h={audio:!0},f=Object(r.useRef)(null),O=[],g=function(){var e=new Blob(O,{type:"audio/ogg; codecs=opus"});O=[];var t=window.URL.createObjectURL(e);i.push({stream:t}),b(i),c("Record")},m=function(){var e=Object(u.a)(s.a.mark((function e(t){return s.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:if(null!==t){e.next=2;break}return e.abrupt("return",navigator.mediaDevices.getUserMedia(h));case 2:case"end":return e.stop()}}),e)})));return function(t){return e.apply(this,arguments)}}();return Object(l.jsxs)(l.Fragment,{children:[Object(l.jsx)("button",{onClick:function(){m(f.current).then((function(e){f.current=new MediaRecorder(e),f.current.onstop=g,f.current.ondataavailable=function(e){O.push(e.data)}})).catch((function(e){console.log("MR already exists")})).then((function(){"Record"===n?(f.current.start(),c("Stop")):f.current.stop()}))},children:n}),i.map((function(e,t){return Object(l.jsx)(j,{stream:e.stream},e.stream.toString())}))]})});n(16);var h=function(){return Object(l.jsxs)(l.Fragment,{children:[Object(l.jsx)("header",{children:Object(l.jsx)("h1",{children:"Sound Recorder"})}),Object(l.jsx)("div",{className:"body",children:Object(l.jsx)(b,{})})]})};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));var f=function(e){e&&e instanceof Function&&n.e(3).then(n.bind(null,18)).then((function(t){var n=t.getCLS,r=t.getFID,c=t.getFCP,o=t.getLCP,a=t.getTTFB;n(e),r(e),c(e),o(e),a(e)}))};a.a.render(Object(l.jsx)(c.a.StrictMode,{children:Object(l.jsx)(h,{})}),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()})).catch((function(e){console.error(e.message)})),f()}],[[17,1,2]]]);
//# sourceMappingURL=main.059d315d.chunk.js.map