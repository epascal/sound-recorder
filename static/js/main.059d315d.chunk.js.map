{"version":3,"sources":["components/Recording/index.js","components/Recorder/index.js","App.js","serviceWorkerRegistration.js","reportWebVitals.js","index.js"],"names":["Recording","stream","controls","src","onClick","toString","Recorder","useState","recordingStateText","setRecordingStateText","recordings","setRecordings","constraints","audio","mediaRecorder","useRef","chunks","onStop","blob","Blob","audioURL","window","URL","createObjectURL","push","initMediaRecorder","mr","a","navigator","mediaDevices","getUserMedia","current","then","currstream","MediaRecorder","onstop","ondataavailable","e","data","catch","err","console","log","start","stop","map","recording","index","App","className","Boolean","location","hostname","match","reportWebVitals","onPerfEntry","Function","getCLS","getFID","getFCP","getLCP","getTTFB","ReactDOM","render","StrictMode","document","getElementById","serviceWorker","ready","registration","unregister","error","message"],"mappings":"ySAwBeA,EArBG,SAAC,GAAgB,IAAdC,EAAa,EAAbA,OAMjB,OACI,mCACA,oCACI,uBAAOC,SAAS,WAAWC,IAAKF,IAChC,kDACA,wBAAQG,QATQ,aAShB,sBAHUH,EAAOI,eC8EdC,G,MApFE,WAAO,IAAD,EACiCC,mBAAS,UAD1C,mBACZC,EADY,KACQC,EADR,OAEiBF,mBAAS,IAF1B,mBAEZG,EAFY,KAEAC,EAFA,KAGbC,EAAc,CAAEC,OAAO,GACvBC,EAAgBC,iBAAO,MACzBC,EAAS,GAEPC,EAAS,WAEX,IAAMC,EAAO,IAAIC,KAAKH,EAAQ,CAAE,KAAS,2BACzCA,EAAS,GACT,IAAMI,EAAWC,OAAOC,IAAIC,gBAAgBL,GAC5CR,EAAWc,KAAK,CAACvB,OAAQmB,IACzBT,EAAcD,GACdD,EAAsB,WAUpBgB,EAAiB,uCAAG,WAAOC,GAAP,SAAAC,EAAA,yDAEX,OAAPD,EAFkB,yCAGXE,UAAUC,aAAaC,aAAalB,IAHzB,2CAAH,sDAgDvB,OACI,qCACI,wBAAQR,QAzCQ,WAEpBqB,EAAkBX,EAAciB,SAC/BC,MAAK,SAACC,GACHnB,EAAciB,QAAU,IAAIG,cAAcD,GAC1CnB,EAAciB,QAAQI,OAASlB,EAC/BH,EAAciB,QAAQK,gBAAkB,SAASC,GAC7CrB,EAAOQ,KAAKa,EAAEC,UAGrBC,OAAM,SAACC,GACJC,QAAQC,IAAI,wBAEfV,MAAK,WACE,WAAaxB,GACbM,EAAciB,QAAQY,QACtBlC,EAAsB,SAEtBK,EAAciB,QAAQa,WAuB1B,SAAmCpC,IAb1BE,EAAWmC,KAAI,SAACC,EAAWC,GACpC,OACI,cAAC,EAAD,CAAW9C,OAAQ6C,EAAU7C,QAAa6C,EAAU7C,OAAOI,oB,MClD5D2C,MAdf,WAEE,OACE,qCACA,iCACE,kDAEF,qBAAKC,UAAU,OAAf,SACE,cAAC,EAAD,UCAcC,QACW,cAA7B7B,OAAO8B,SAASC,UAEe,UAA7B/B,OAAO8B,SAASC,UAEhB/B,OAAO8B,SAASC,SAASC,MAAM,2DCjBnC,IAYeC,EAZS,SAAAC,GAClBA,GAAeA,aAAuBC,UACxC,6BAAqBxB,MAAK,YAAkD,IAA/CyB,EAA8C,EAA9CA,OAAQC,EAAsC,EAAtCA,OAAQC,EAA8B,EAA9BA,OAAQC,EAAsB,EAAtBA,OAAQC,EAAc,EAAdA,QAC3DJ,EAAOF,GACPG,EAAOH,GACPI,EAAOJ,GACPK,EAAOL,GACPM,EAAQN,OCAdO,IAASC,OACP,cAAC,IAAMC,WAAP,UACE,cAAC,EAAD,MAEFC,SAASC,eAAe,SFoHpB,kBAAmBtC,WACrBA,UAAUuC,cAAcC,MACrBpC,MAAK,SAACqC,GACLA,EAAaC,gBAEd/B,OAAM,SAACgC,GACN9B,QAAQ8B,MAAMA,EAAMC,YE/G5BlB,M","file":"static/js/main.059d315d.chunk.js","sourcesContent":["import './style.css'\n\n\nconst Recording = ({ stream }) => {\n\n    const deleteRecording = () => {\n\n    }\n\n    return (\n        <>\n        <article key={stream.toString()}>\n            <audio controls=\"controls\" src={stream}></audio>\n            <p>Unnamed recording</p>\n            <button onClick={deleteRecording}>Delete</button>\n        </article>\n        </>\n    );\n}\n\nRecording.propTypes = {\n\n}\n\nexport default Recording;","import { useState, useRef } from 'react'\n//import Visualizer from '../Visualizer'\nimport Recording from '../Recording'\nimport './style.css'\n\nconst Recorder = () => {\n    const [recordingStateText, setRecordingStateText] = useState('Record')\n    const [recordings, setRecordings] = useState([])\n    const constraints = { audio: true }\n    const mediaRecorder = useRef(null)\n    let chunks = []\n\n    const onStop = () => {\n\n        const blob = new Blob(chunks, { 'type' : 'audio/ogg; codecs=opus' });\n        chunks = [];\n        const audioURL = window.URL.createObjectURL(blob);\n        recordings.push({stream: audioURL})\n        setRecordings(recordings)\n        setRecordingStateText('Record')\n        \n    }\n\n    /**\n     * We only want to initialize if one doesn't already exist\n     * \n     * @param {MediaRecorder} mr \n     * @returns \n     */\n    const initMediaRecorder = async (mr) => {\n\n        if( mr === null ) {\n            return navigator.mediaDevices.getUserMedia(constraints)\n        }\n    \n    }\n    \n\n    const toggleRecording = () => {\n        \n        initMediaRecorder(mediaRecorder.current)\n        .then((currstream) => {\n            mediaRecorder.current = new MediaRecorder(currstream);\n            mediaRecorder.current.onstop = onStop\n            mediaRecorder.current.ondataavailable = function(e) {\n                chunks.push(e.data);\n            }\n        })\n        .catch((err) => {\n            console.log('MR already exists')\n        })\n        .then(() => {\n            if( 'Record' === recordingStateText ) {\n                mediaRecorder.current.start();\n                setRecordingStateText('Stop')\n            } else {\n                mediaRecorder.current.stop();\n            }\n        })\n\n        \n    }\n\n\n    const renderAudio = () => {\n\n        let audios = recordings.map((recording, index) => {\n            return (\n                <Recording stream={recording.stream} key={recording.stream.toString()} />\n            )   \n        })\n        \n        return audios\n\n    }\n\n\n    return (\n        <>\n            <button onClick={toggleRecording}>{recordingStateText}</button>\n            {renderAudio()}\n        </>\n    )\n}\n\nRecorder.propTypes = {\n    //stream: PropTypes.node\n};\n\nexport default Recorder;\n","import Recorder from './components/Recorder'\nimport './App.css';\n\n\nfunction App() {\n\n  return (\n    <>\n    <header>\n      <h1>Sound Recorder</h1>\n    </header>\n    <div className=\"body\">\n      <Recorder />\n    </div>\n    </>\n  );\n}\n\nexport default App;\n","// This optional code is used to register a service worker.\n// register() is not called by default.\n\n// This lets the app load faster on subsequent visits in production, and gives\n// it offline capabilities. However, it also means that developers (and users)\n// will only see deployed updates on subsequent visits to a page, after all the\n// existing tabs open on the page have been closed, since previously cached\n// resources are updated in the background.\n\n// To learn more about the benefits of this model and instructions on how to\n// opt-in, read https://cra.link/PWA\n\nconst isLocalhost = Boolean(\n  window.location.hostname === 'localhost' ||\n    // [::1] is the IPv6 localhost address.\n    window.location.hostname === '[::1]' ||\n    // 127.0.0.0/8 are considered localhost for IPv4.\n    window.location.hostname.match(/^127(?:\\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/)\n);\n\nexport function register(config) {\n  if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {\n    // The URL constructor is available in all browsers that support SW.\n    const publicUrl = new URL(process.env.PUBLIC_URL, window.location.href);\n    if (publicUrl.origin !== window.location.origin) {\n      // Our service worker won't work if PUBLIC_URL is on a different origin\n      // from what our page is served on. This might happen if a CDN is used to\n      // serve assets; see https://github.com/facebook/create-react-app/issues/2374\n      return;\n    }\n\n    window.addEventListener('load', () => {\n      const swUrl = `${process.env.PUBLIC_URL}/service-worker.js`;\n\n      if (isLocalhost) {\n        // This is running on localhost. Let's check if a service worker still exists or not.\n        checkValidServiceWorker(swUrl, config);\n\n        // Add some additional logging to localhost, pointing developers to the\n        // service worker/PWA documentation.\n        navigator.serviceWorker.ready.then(() => {\n          console.log(\n            'This web app is being served cache-first by a service ' +\n              'worker. To learn more, visit https://cra.link/PWA'\n          );\n        });\n      } else {\n        // Is not localhost. Just register service worker\n        registerValidSW(swUrl, config);\n      }\n    });\n  }\n}\n\nfunction registerValidSW(swUrl, config) {\n  navigator.serviceWorker\n    .register(swUrl)\n    .then((registration) => {\n      registration.onupdatefound = () => {\n        const installingWorker = registration.installing;\n        if (installingWorker == null) {\n          return;\n        }\n        installingWorker.onstatechange = () => {\n          if (installingWorker.state === 'installed') {\n            if (navigator.serviceWorker.controller) {\n              // At this point, the updated precached content has been fetched,\n              // but the previous service worker will still serve the older\n              // content until all client tabs are closed.\n              console.log(\n                'New content is available and will be used when all ' +\n                  'tabs for this page are closed. See https://cra.link/PWA.'\n              );\n\n              // Execute callback\n              if (config && config.onUpdate) {\n                config.onUpdate(registration);\n              }\n            } else {\n              // At this point, everything has been precached.\n              // It's the perfect time to display a\n              // \"Content is cached for offline use.\" message.\n              console.log('Content is cached for offline use.');\n\n              // Execute callback\n              if (config && config.onSuccess) {\n                config.onSuccess(registration);\n              }\n            }\n          }\n        };\n      };\n    })\n    .catch((error) => {\n      console.error('Error during service worker registration:', error);\n    });\n}\n\nfunction checkValidServiceWorker(swUrl, config) {\n  // Check if the service worker can be found. If it can't reload the page.\n  fetch(swUrl, {\n    headers: { 'Service-Worker': 'script' },\n  })\n    .then((response) => {\n      // Ensure service worker exists, and that we really are getting a JS file.\n      const contentType = response.headers.get('content-type');\n      if (\n        response.status === 404 ||\n        (contentType != null && contentType.indexOf('javascript') === -1)\n      ) {\n        // No service worker found. Probably a different app. Reload the page.\n        navigator.serviceWorker.ready.then((registration) => {\n          registration.unregister().then(() => {\n            window.location.reload();\n          });\n        });\n      } else {\n        // Service worker found. Proceed as normal.\n        registerValidSW(swUrl, config);\n      }\n    })\n    .catch(() => {\n      console.log('No internet connection found. App is running in offline mode.');\n    });\n}\n\nexport function unregister() {\n  if ('serviceWorker' in navigator) {\n    navigator.serviceWorker.ready\n      .then((registration) => {\n        registration.unregister();\n      })\n      .catch((error) => {\n        console.error(error.message);\n      });\n  }\n}\n","const reportWebVitals = onPerfEntry => {\n  if (onPerfEntry && onPerfEntry instanceof Function) {\n    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {\n      getCLS(onPerfEntry);\n      getFID(onPerfEntry);\n      getFCP(onPerfEntry);\n      getLCP(onPerfEntry);\n      getTTFB(onPerfEntry);\n    });\n  }\n};\n\nexport default reportWebVitals;\n","import React from 'react';\nimport ReactDOM from 'react-dom';\nimport './index.css';\nimport App from './App';\nimport * as serviceWorkerRegistration from './serviceWorkerRegistration';\nimport reportWebVitals from './reportWebVitals';\n\nReactDOM.render(\n  <React.StrictMode>\n    <App />\n  </React.StrictMode>,\n  document.getElementById('root')\n);\n\n// If you want your app to work offline and load faster, you can change\n// unregister() to register() below. Note this comes with some pitfalls.\n// Learn more about service workers: https://cra.link/PWA\nserviceWorkerRegistration.unregister();\n\n// If you want to start measuring performance in your app, pass a function\n// to log results (for example: reportWebVitals(console.log))\n// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals\nreportWebVitals();\n"],"sourceRoot":""}