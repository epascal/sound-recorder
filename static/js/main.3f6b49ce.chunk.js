(this["webpackJsonpsound-recorder"]=this["webpackJsonpsound-recorder"]||[]).push([[0],[,,,,,,,,,,,,,,function(e,t,n){},,function(e,t,n){},,function(e,t,n){},function(e,t,n){},function(e,t,n){},function(e,t,n){"use strict";n.r(t);var r=n(1),o=n.n(r),c=n(8),a=n.n(c),i=(n(14),n(7)),s=n.n(i),u=n(9),d=n(2),l=n(3),f=(n(16),n(0)),j=function(e){var t=e.stream,n=e.name,r=e.onDeleteHandler,o=e.onEditNameHandler,c=e.id;return Object(f.jsxs)("article",{id:c,children:[Object(f.jsx)("audio",{controls:"controls",src:t,preload:"auto",role:"application",children:"Sorry, your browser doesn't support recording audio."}),Object(f.jsxs)("p",{children:[Object(f.jsx)("span",{className:"name",role:"presentation",children:n}),Object(f.jsx)("button",{onClick:function(e){o(e)},className:"editName",title:"Click to edit name",children:"\u270f\ufe0f"})]}),Object(f.jsx)("button",{onClick:function(e){r(e)},className:"delete",children:"Delete"})]})};j.defaultProps={stream:{stream:"blob:http://localhost",name:"Default recording name",id:"id0"}};var b=j,m=(n(18),function(e){var t,n,o,c,a,i,s,u=e.stream,d=e.barColor,l=void 0===d?[0,0,0]:d,j=Object(r.useRef)();Object(r.useEffect)((function(){o=j.current.getContext("2d"),t=j.current,window.onresize=function(){t.width=document.querySelector("body").offsetWidth},window.onresize(),b(u)}),[]);var b=function(e){n||(n=new(window.AudioContext||window.webkitAudioContext));var t=n.createMediaStreamSource(e);(c=n.createAnalyser()).fftSize=256,i=c.frequencyBinCount,a=new Uint8Array(i),t.connect(c),window.requestAnimationFrame(m)},m=function e(n){if(s!==n){var r=t.width,u=t.height;c.getByteTimeDomainData(a);var d,f=r/i,j=0;o.clearRect(0,0,r,u);for(var b=0;b<i;b++)d=a[b],o.fillStyle="rgb(".concat(l[0],", ").concat(l[1],", ").concat(l[2],")"),o.fillRect(j,u,f,0-d/2),j+=f+1;s=n}window.requestAnimationFrame(e)};return Object(f.jsx)(f.Fragment,{children:Object(f.jsx)("canvas",{ref:j,className:"visualizer"})})});n(19);var O=function(e){var t=e.stream,n=function(e){var t=Object(r.useState)(!1),n=Object(d.a)(t,2),o=n[0],c=n[1],a=Object(r.useState)([]),i=Object(d.a)(a,2),s=i[0],u=i[1],f=Object(r.useState)([]),j=Object(d.a)(f,2),b=j[0],m=j[1],O=Object(r.useMemo)((function(){return new MediaRecorder(e)}),[e]);return O.onstop=function(){var e=new Blob(b,{type:"audio/ogg; codecs=opus"}),t=window.URL.createObjectURL(e);u((function(e){return[].concat(Object(l.a)(e),[{stream:t,name:(new Date).toISOString().split(".")[0].split("T").join(" "),id:"id".concat(window.performance.now().toString())}])})),m([]),c(!1)},O.onstart=function(){c(!0)},O.ondataavailable=function(e){m((function(t){return[].concat(Object(l.a)(t),[e.data])}))},{recorder:O,recordings:s,setRecordings:u,isRecording:o}}(t),o=n.recorder,c=n.recordings,a=n.setRecordings,i=n.isRecording,s="record-play",u=Object(r.useMemo)((function(){return i?"".concat(s," recording-audio"):s}),[i]),j=Object(r.useMemo)((function(){return i?"Stop":"Record"}),[i]),O=function(e){var t,n=e.target.parentNode.parentNode.attributes.id.value,r=Object(l.a)(c),o=c.filter((function(e){return e.id===n&&e})),i=c.indexOf(o[0]),s=null!==(t=window.prompt("Enter a new name",o[0].name))&&void 0!==t?t:o[0].name;o[0].name=s,r.splice(i,1,o[0]),a(r)},p=function(e){var t=e.target.parentNode.attributes.id.value,n=window.confirm("Are you sure you want to delete this recording?");if(!0===n){var r=c.filter((function(e){return t!==e.id}));e.target.parentNode.classList.add("vanish"),setTimeout((function(){a(Object(l.a)(r))}),900)}};return Object(f.jsxs)(f.Fragment,{children:[Object(f.jsx)(m,{stream:t,isRecording:i,barColor:[18,124,85]}),Object(f.jsx)("button",{onClick:function(){i?o.stop():o.start(1e3)},className:u,children:j}),Object(f.jsx)("section",{children:c.map((function(e,t){return Object(f.jsx)(b,{stream:e.stream,name:e.name,id:e.id,onDeleteHandler:p,onEditNameHandler:O},e.id)}))})]})};n(20);var p=function(){var e=Object(r.useMemo)((function(){return{audio:!0}}),[]),t=Object(r.useState)(null),n=Object(d.a)(t,2),o=n[0],c=n[1],a=Object(r.useState)(null),i=Object(d.a)(a,2),l=i[0],j=i[1];return Object(r.useEffect)((function(){if(!o){var t=!1;return function(){var n=Object(u.a)(s.a.mark((function n(){var r;return s.a.wrap((function(n){for(;;)switch(n.prev=n.next){case 0:return n.prev=0,n.next=3,navigator.mediaDevices.getUserMedia(e);case 3:r=n.sent,t||c(r),n.next=10;break;case 7:n.prev=7,n.t0=n.catch(0),t||j(n.t0);case 10:case"end":return n.stop()}}),n,null,[[0,7]])})));return function(){return n.apply(this,arguments)}}()(),function(){t=!0,o&&(o.getAudioTracks&&o.getAudioTracks().map((function(e){return e.stop()})),o.stop&&o.stop())}}}),[e,o,l]),Object(f.jsxs)(f.Fragment,{children:[Object(f.jsx)("header",{children:Object(f.jsx)("h1",{children:"Sound Recorder"})}),Object(f.jsx)("main",{children:null===o?Object(f.jsx)("button",{className:"record-play",children:"Loading\u2026"}):Object(f.jsx)(O,{stream:o})})]})};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));var h=function(e){e&&e instanceof Function&&n.e(3).then(n.bind(null,22)).then((function(t){var n=t.getCLS,r=t.getFID,o=t.getFCP,c=t.getLCP,a=t.getTTFB;n(e),r(e),o(e),c(e),a(e)}))};a.a.render(Object(f.jsx)(o.a.StrictMode,{children:Object(f.jsx)(p,{})}),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()})).catch((function(e){console.error(e.message)})),h()}],[[21,1,2]]]);
//# sourceMappingURL=main.3f6b49ce.chunk.js.map