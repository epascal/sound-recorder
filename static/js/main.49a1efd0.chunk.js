(this["webpackJsonpsound-recorder"]=this["webpackJsonpsound-recorder"]||[]).push([[0],[,,,,,,,,,,,function(e,t,n){},,function(e,t,n){},,function(e,t,n){},function(e,t,n){},function(e,t,n){"use strict";n.r(t);var c=n(1),r=n.n(c),o=n(5),a=n.n(o),i=(n(11),n(4)),s=n.n(i),u=n(6),d=n(2),l=(n(13),n(0)),j=function(e){var t=e.stream;return Object(l.jsx)(l.Fragment,{children:Object(l.jsxs)("article",{children:[Object(l.jsx)("audio",{controls:"controls",src:t}),Object(l.jsxs)("p",{children:[Object(l.jsx)("span",{className:"name",children:"Unnamed recording"}),Object(l.jsx)("button",{onClick:function(){},className:"editName",title:"Click to edit name",children:"\u270f\ufe0f"})]}),Object(l.jsx)("button",{onClick:function(){},className:"delete",children:"Delete"})]},t.toString())})},b=(n(15),function(){var e=Object(c.useState)("Record"),t=Object(d.a)(e,2),n=t[0],r=t[1],o=Object(c.useState)("record-play"),a=Object(d.a)(o,2),i=a[0],b=a[1],h=Object(c.useState)([]),f=Object(d.a)(h,2),m=f[0],O=f[1],p={audio:!0},g=Object(c.useRef)(null),x=[],v=[i],w=function(){var e=new Blob(x,{type:"audio/ogg; codecs=opus"});x=[];var t=window.URL.createObjectURL(e);m.push({stream:t}),v.pop(),b(v.join(" ")),O(m),r("Record")},k=function(){v.push("recording-audio"),console.log({recordButtonClasses:v}),b(v.join(" ")),r("Stop")},R=function(){var e=Object(u.a)(s.a.mark((function e(t){return s.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:if(null!==t){e.next=2;break}return e.abrupt("return",navigator.mediaDevices.getUserMedia(p));case 2:case"end":return e.stop()}}),e)})));return function(t){return e.apply(this,arguments)}}();return Object(l.jsxs)(l.Fragment,{children:[Object(l.jsx)("button",{onClick:function(){R(g.current).then((function(e){g.current=new MediaRecorder(e),g.current.onstop=w,g.current.onstart=k,g.current.ondataavailable=function(e){x.push(e.data)}})).catch((function(e){console.log("MR already exists")})).then((function(){"Record"===n?g.current.start():g.current.stop()}))},className:i,children:n}),m.map((function(e,t){return Object(l.jsx)(j,{stream:e.stream},e.stream.toString())}))]})});n(16);var h=function(){return Object(l.jsxs)(l.Fragment,{children:[Object(l.jsx)("header",{children:Object(l.jsx)("h1",{children:"Sound Recorder"})}),Object(l.jsx)("main",{children:Object(l.jsx)(b,{})})]})};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));var f=function(e){e&&e instanceof Function&&n.e(3).then(n.bind(null,18)).then((function(t){var n=t.getCLS,c=t.getFID,r=t.getFCP,o=t.getLCP,a=t.getTTFB;n(e),c(e),r(e),o(e),a(e)}))};a.a.render(Object(l.jsx)(r.a.StrictMode,{children:Object(l.jsx)(h,{})}),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()})).catch((function(e){console.error(e.message)})),f()}],[[17,1,2]]]);
//# sourceMappingURL=main.49a1efd0.chunk.js.map