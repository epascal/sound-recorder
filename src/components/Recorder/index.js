import { useState } from 'react'
import useMediaRecorder from '../../hooks/useMediaRecorder'
//import Visualizer from '../Visualizer'
import Recording from '../Recording'
import './style.css'

const Recorder = ({stream}) => {
    const [recordingStateText, setRecordingStateText] = useState('Record')
    const [recordings, setRecordings] = useState([])
    const [recordButtonClassesText, setRecordButtonClassesText] = useState('record-play')
    let chunks = []

    const onStop = () => {

        const blob = new Blob(chunks, { 'type' : 'audio/ogg; codecs=opus' });
        chunks = [];
        const audioURL = window.URL.createObjectURL(blob);
        recordings.push({stream: audioURL})
        setRecordings(recordings)

    }


    const onStart = () => {

    
    }


    const ondataavailable = (e) => {
        chunks.push(e.data);
    }


    const toggleRecording = () => {

        if( 'Record' === recordingStateText ) {
            setRecordButtonClassesText(recordButtonClassesText + ' recording-audio')
            setRecordingStateText('Stop')
            mediaRecorder.start();
        } else {
            setRecordButtonClassesText(recordButtonClassesText.split(' ').reverse().pop().toString())
            setRecordingStateText('Record')
            mediaRecorder.stop();
        }

    }

    const mediaRecorder = useMediaRecorder(stream, recordButtonClassesText, {onStart: onStart, onStop: onStop, ondataavailable: ondataavailable})

    const renderAudio = () => {

        let audios = recordings.map((recording, index) => {
            return (
                <Recording stream={recording.stream} key={recording.stream.toString()} />
            )   
        })
        
        return audios

    }


    return (
        <>
            <button onClick={toggleRecording} className={recordButtonClassesText}>{recordingStateText}</button>
            {renderAudio()}
        </>
    )
}

Recorder.propTypes = {
    //stream: PropTypes.node
};

export default Recorder
