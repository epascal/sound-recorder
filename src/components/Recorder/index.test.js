import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import '../../mocks/mediaRecorder.js';
import Recorder from './index'

beforeEach(() => {
    // Clear all instances and calls to constructor and all methods:
    // Recorder.mockClear();
});

/**
 * We need to mock the getUserMedia function
 * https://github.com/goldingdamien/get-user-media-mock
 * https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
 */
it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Recorder />, div);
})

it('record turns colors when clicked', async () => {
    // const useRefSpy = jest.spyOn(React, 'useRef').mockReturnValueOnce({current: {MediaRecorder}})
    // const spy = jest.spyOn(MediaRecorder, 'start')
    const component = await renderer.create(
        <Recorder />,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    
      // manually trigger the callback
      tree.props.onClick();
      // re-rendering
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    
      // manually trigger the callback
      tree.props.onClick();
      // re-rendering
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    //   spy.mockRestore();
})

it('prompts for name when recording is done', () => {
    // click recording button to start recording
    // click recording button to stop recording
    // expect to see name dialog
})

it('saves name', () => {

})