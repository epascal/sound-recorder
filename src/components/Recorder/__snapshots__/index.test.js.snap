// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`record turns colors when clicked 1`] = `
<button
  className="record-play"
  onClick={[Function]}
>
  Record
</button>
`;

exports[`record turns colors when clicked 2`] = `
<button
  className="record-play"
  onClick={[Function]}
>
  Record
</button>
`;

exports[`record turns colors when clicked 3`] = `
<button
  className="record-play"
  onClick={[Function]}
>
  Record
</button>
`;
